ocaml-lsp-server/src/document_store.mli

```diff
@@ -2,16 +2,22 @@ open Import
 
 type t
 
-val make : unit -> t
+val make : _ Server.t Fdecl.t -> Fiber.Pool.t -> t
 
-val put : t -> Document.t -> unit
+val open_ : t -> Document.t -> unit Fiber.t
```

Comment:
I would say `put` had a clearer intent. With `open_` it's slightly confusing in what way the file is opened

Reply:
`put` had a clearer intent but its old semantics are gone with the new implementation. The new function only corresponds to opening a new document, so I don't know of a better name. Certainly `put` isn't any better. Do you have another suggestion?

Reply:
maybe `on_doc_open`?

`open_` is also not so ideal because 
1. There's `close` fn but it closes the store rather than the doc
2. The reverse of `open_` is named `remove_document`, which isn't too obvious. 

Anyway, as you prefer